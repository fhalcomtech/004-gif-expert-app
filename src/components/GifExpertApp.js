import React, { useState } from 'react';
import { AddCategoryComp } from './AddCategoryComp';
import { ShowCategories } from './ShowCategories';

const GifExpertApp = ({defaultCategories= []}) =>
{
  const [categ, setCateg] = useState(defaultCategories);
  return (
    <>
      <AddCategoryComp setCategories={setCateg} />
      <hr/>
      <div>
         <ul>{categ.map(ct=><ShowCategories key={ct} category={ct} />)}</ul>
      </div>
    </>
  )

  
}

export default GifExpertApp;