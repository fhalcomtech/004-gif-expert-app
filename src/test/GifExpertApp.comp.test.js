import {shallow} from "enzyme";
import GifExpertApp from "../components/GifExpertApp";

describe('Test for GifExpertApp component', () => {
    test('Test Initial Snapshot without default categories', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    })

    test('Test Initial Snapshot with initial categories', (done) => {
        const defaultCategories = ['goku','gohan'];
        const wrapper = shallow(<GifExpertApp defaultCategories={defaultCategories}/>);
        expect(wrapper.find('ShowCategories').length).toBe(defaultCategories.length)
        expect(wrapper).toMatchSnapshot();
        done();
    })
});