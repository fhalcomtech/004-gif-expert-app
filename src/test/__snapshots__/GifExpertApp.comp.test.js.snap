// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for GifExpertApp component Test Initial Snapshot with initial categories 1`] = `
<Fragment>
  <AddCategoryComp
    setCategories={[Function]}
  />
  <hr />
  <div>
    <ul>
      <ShowCategories
        category="goku"
        key="goku"
      />
      <ShowCategories
        category="gohan"
        key="gohan"
      />
    </ul>
  </div>
</Fragment>
`;

exports[`Test for GifExpertApp component Test Initial Snapshot without default categories 1`] = `
<Fragment>
  <AddCategoryComp
    setCategories={[Function]}
  />
  <hr />
  <div>
    <ul />
  </div>
</Fragment>
`;
